alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -563,15 +563,37 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         return hits.map { hit ->
             val sourceMap = hit.sourceAsMap
 
-            var xContentBuilder = XContentFactory.jsonBuilder().startObject()
-            sourceMap.forEach { (k, v) ->
-                xContentBuilder = xContentBuilder.field("${k}_${index}_$monitorId", v)
-            }
-            xContentBuilder = xContentBuilder.endObject()
+            transformDocumentFieldNames(sourceMap, "${index}_$monitorId")
+
+            var xContentBuilder = XContentFactory.jsonBuilder().map(sourceMap)
 
             val sourceRef = BytesReference.bytes(xContentBuilder)
 
             Pair(hit.id, sourceRef)
         }
     }
+
+    /**
+     * Traverses document fields in leaves recursively and appends [fieldNameSuffix] to field names.
```

Comment:
mention that we are trying to flatten the object type fields

Reply:
Actually, we can't flatten it, because of type=nested. You can't insert document with flatten fields if type is nested. For example, let's say field "rule" is type=nested, then this is invalid:
```
{
     "rule.category": "windows"
}
```
but this is valid:
```
{
     "rule" : {
          "category": "windows"
     }
}
```

Reply:
i am not talking about type=nested